src/functions/href.js

```diff
@@ -54,6 +54,8 @@ defineFunction({
             if (first === last) { // Case 2 : type of both ends coincides
                 classes = [first];
             } else { // Case 3: both ends have different types.
+                // TODO(kevinb): figure out a better way to communicate this
+                // information to buildHTML.js#buildExpression.
```

Comment:
I think we can just `return new buildCommon.makeAnchor(href, [], elements, options);` here

Reply:
I was confused at first, but this makes sense given your comments in `getTypeOfDomTree`.

Reply:
Currently spacings around `\href` are duplicated as dummy spans are created.